include/picolibrary/iterator.h

```diff
@@ -136,6 +137,477 @@ struct Iterator_Traits<T *> {
+     *        position..
+     *
+     * \param[in] offset The offset of the element to access.
+     *
+     * \return The element at the specified offset from the current iterator position.
+     */
+    constexpr auto operator[]( Offset offset ) const noexcept -> Reference
+    {
+        return m_iterator[ -offset - 1 ];
+    }
+
+    /**
+     * \brief Advance the iterator to the next element.
+     *
+     * \return The advanced iterator.
+     */
+    constexpr auto & operator++() noexcept
+    {
+        --m_iterator;
+
+        return *this;
+    }
+
+    /**
+     * \brief Retreat the iterator to the previous element.
+     *
+     * \return The retreated iterator.
+     */
+    constexpr auto & operator--() noexcept
+    {
+        ++m_iterator;
+
+        return *this;
+    }
+
+    /**
+     * \brief Advance the iterator to the next element.
+     *
+     * \return A copy of the iterator that was made before the iterator was advanced.
+     */
+    constexpr auto operator++( int ) noexcept
+    {
+        auto iterator = m_iterator;
+
+        --m_iterator;
+
+        return iterator;
+    }
+
+    /**
+     * \brief Retreat the iterator to the previous element.
+     *
+     * \return A copy of the iterator that was made before the iterator was retreated.
+     */
+    constexpr auto operator--( int ) noexcept
+    {
+        auto iterator = m_iterator;
+
+        ++m_iterator;
+
+        return iterator;
+    }
+
+    /**
+     * \brief Advance a copy of the iterator n positions.
+     *
+     * \param[in] n The number of positions to advance the iterator.
+     *
+     * \return A copy of the iterator that has been advanced n positions.
+     */
+    constexpr auto operator+( Offset n ) const noexcept
+    {
+        return Reverse_Iterator{ m_iterator - n };
+    }
+
+    /**
+     * \brief Retreat a copy of the iterator n positions.
+     *
+     * \param[in] n The number of positions to retreat the iterator.
+     *
+     * \return A copy of the iterator that has been retreated n positions.
+     */
+    constexpr auto operator-( Offset n ) const noexcept
+    {
+        return Reverse_Iterator{ m_iterator + n };
+    }
+
+    /**
+     * \brief Advance the iterator n positions.
+     *
+     * \param[in] n The number of positions to advance the iterator.
+     *
+     * \return The advanced iterator.
+     */
+    constexpr auto & operator+=( Offset n ) noexcept
+    {
+        m_iterator -= n;
+
+        return *this;
+    }
+
+    /**
+     * \brief Retreat the iterator n positions.
+     *
+     * \param[in] n The number of positions to retreat the iterator.
+     *
+     * \return The retreated iterator.
+     */
+    constexpr auto & operator-=( Offset n ) noexcept
+    {
+        m_iterator += n;
+
+        return *this;
+    }
+
+  protected:
+    /**
+     * \brief The adapted iterator.
+     */
+    Iterator m_iterator{};
+};
+
+/**
+ * \brief Build a reverse iterator.
```

Comment:
Tie this function to `::picolibrary::Reverse_Iterator`'s documentation (`\relatedalso picolibrary::Reverse_Iterator`).